producer-c/producer-cloudwatch-integ/KvsVideoOnlyStreamingSampleCloudwatch.cpp

```diff
@@ -0,0 +1,228 @@
+#include <com/amazonaws/kinesis/video/cproducer/Include.h>
+#include <com/amazonaws/kinesis/video/utils/Include.h>
+#include "CanaryStreamCallbacks.h"
+
+volatile ATOMIC_BOOL sigCaptureInterrupt;
+
+#define DEFAULT_RETENTION_PERIOD            (2 * HUNDREDS_OF_NANOS_IN_AN_HOUR)
+#define DEFAULT_BUFFER_DURATION             (120 * HUNDREDS_OF_NANOS_IN_A_SECOND)
+#define DEFAULT_CALLBACK_CHAIN_COUNT        5
+#define DEFAULT_KEY_FRAME_INTERVAL          45
+#define DEFAULT_FPS_VALUE                   25
+#define DEFAULT_STREAM_DURATION             (20 * HUNDREDS_OF_NANOS_IN_A_SECOND)
+
+#define NUMBER_OF_FRAME_FILES               403
+#define CANARY_METADATA_SIZE                (SIZEOF(INT64) + SIZEOF(UINT32) + SIZEOF(UINT32) + SIZEOF(UINT64))
+
+#define CANARY_FILE_LOGGING_BUFFER_SIZE     (200 * 1024)
+#define CANARY_MAX_NUMBER_OF_LOG_FILES      10
+VOID sigintHandler(INT32 sigNum)
+{
+	ATOMIC_STORE_BOOL(&sigCaptureInterrupt, TRUE);
+}
+
+// add frame pts, frame index, original frame size, CRC to beginning of buffer
+VOID addCanaryMetadataToFrameData(PFrame pFrame) {
+    putInt64((PINT64)pFrame->frameData, pFrame->presentationTs / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
```

Comment:
Might want to do an aligned versions of these

Reply:
Also, arranging bits in a buffer we usually do with a pointer.  Ex:

PBYTE pCurPtr = blah;

PUT_UNALIGNED_INT32(pCurPtr, something);
pCurPtr += SIZEOF(UINT32);

...

Reply:
Done